validators/BudgetFields.ts

```diff
@@ -0,0 +1,16 @@
+import { z } from 'zod';
+
+export const BudgetSch = z.object({
+  name: z.string(),
+  description: z.string(),
+  budgetType: z.enum(['static', 'progressive']),
+  estimatedCost: z.number(),
+  // Zod docs recommended way of accepting dates as strings
+  estimatedRealizationDate: z.preprocess((arg) => {
+    if (typeof arg == 'string' || arg instanceof Date) return new Date(arg);
+  }, z.date()),
```

Comment:
Is it in ISOString format?

Reply:
Date string from input is passed into Date constructor and it works with the DB, tested.